src/plugins/expressions/common/expression_functions/specs/map_column.ts

```diff
@@ -101,7 +101,18 @@ export const mapColumn: ExpressionFunctionDefinition<
     });
 
     return Promise.all(rowPromises).then((rows) => {
-      const existingColumnIndex = columns.findIndex(({ name }) => name === args.name);
+      const existingColumnIndex = columns.findIndex(({ id, name }) => {
+        // Columns that have IDs are allowed to have duplicate names, for example esaggs
+        if (args.id) {
+          return id === args.id;
```

Comment:
If the `args.id` is passed but the column does not have `id` is it ok to return `false`?
I didn't see any test about this. I assume it's ok, but probably it's better to make it clear somewhere (either with a test or documentation).

Reply:
@dej611 good catch, I thought I had a test for this but I will add one.